homeassistant/components/dsmr/config_flow.py

```diff
@@ -55,9 +148,26 @@ async def async_step_import(self, import_config=None):
         if status is not None:
             return status
 
+        try:
+            info = await _validate_dsmr_connection(self.hass, import_config)
+        except CannotConnect:
+            return self.async_abort(reason="cannot_connect")
+        except CannotCommunicate:
+            return self.async_abort(reason="cannot_communicate")
+
         if host is not None:
             name = f"{host}:{port}"
         else:
             name = port
 
-        return self.async_create_entry(title=name, data=import_config)
+        data = {**import_config, **info}
```

Comment:
Can we use the serial numbers to set a unique id for the config entry?

Reply:
Yes, I guess we should then also abort if unique id already exists.